nni/compression/pytorch/compressor.py

```diff
@@ -433,6 +432,21 @@ def load_model_state_dict(self, model_state):
         else:
             self.bound_model.load_state_dict(model_state)
 
+    def get_pruned_weights(self, dim=0):
+        for _, wrapper in enumerate(self.get_modules_wrapper()):
+            weight_mask = wrapper.weight_mask
+            mask_size = weight_mask.size()
+            if len(mask_size) == 4:
+                sum_idx = [0, 1, 2, 3]
+                sum_idx.remove(dim)
+                index = torch.nonzero(weight_mask.abs().sum(sum_idx) != 0).tolist()
+            elif len(mask_size) == 2:
+                index = torch.nonzero(weight_mask.abs().sum(1) != 0).tolist()
+            elif len(mask_size) == 1:
```

Comment:
why specifically deal with 4, 2, 1. what about other values? how to deal with them?